### PR TITLE
Feature/nex 19 adapt the test item previewer

### PR DESCRIPTION
**Pull request by krampstudio**
Related to https://oat-sa.atlassian.net/browse/NEX-19

Adapt the test/item previewer to the new initialization format.
Migrate the extension to ES2015 by the way.

Requires : 
 - [x] https://github.com/oat-sa/extension-tao-test/pull/289

How to test : 
- View some results
- Click the "review" button, it should open a test runner to preview an item

Run 
```
cd tao/views/build
npx grunt connect:test taoqtitestpreviewertest
```

### CODE REVIEW

File: scripts/update/Updater.php [hunk 1]
```diff
@@ -87,6 +87,6 @@ public function update($initialVersion)
             $this->setVersion('0.2.0');
         }
 
-        $this->skip('0.2.0', '2.2.1');
+        $this->skip('0.2.0', '3.0.0');
```
Comment (btamas) [inline scripts/update/Updater.php]: This one or the version in manifest.php is not correct.

File: views/js/previewer/adapter/item/qtiItem.js [hunk 2]
```diff
@@ -74,22 +64,48 @@ define([
...
+                testDefinition: 'test-container',
+                serviceCallId: 'previewer',
+                providers: {
+                    runner: {
+                        id: 'qtiItemPreviewer',
+                        module: 'taoQtiTestPreviewer/previewer/provider/item/item',
+                        bundle: 'taoQtiTestPreviewer/loader/qtiPreviewer.min',
+                        category: 'previewer'
+                    },
+                    "proxy": {
```
Comment (btamas) [inline views/js/previewer/adapter/item/qtiItem.js]: What is the reason behind mixing quoted key names and unquoted key names? I think you should not quote them.

File: views/js/previewer/proxy/item.js [hunk 3]
```diff
@@ -27,10 +27,9 @@ define([
     'i18n',
     'core/promise',
```
Comment (btamas) [inline views/js/previewer/proxy/item.js]: You should remove promise.

File: views/js/previewer/runner.js [hunk 4]
```diff
@@ -31,43 +31,29 @@ define([
 ], function ($, _, context, Promise, runnerComponentFactory, runnerTpl) {
```
Comment (btamas) [inline views/js/previewer/runner.js]: `context` and `promise` are not necessary.

File: views/js/test/previewer/plugins/controls/close/test.js [hunk 5]
```diff
@@ -38,6 +38,36 @@ define([
 ) {
     'use strict';
 
+    const runnerConfig = {
+        serviceCallId : 'foo',
+        providers : {
+            runner: {
+                'id': 'qtiItemPreviewer',
```
Comment (btamas) [inline views/js/test/previewer/plugins/controls/close/test.js]: You should not quote keys.

File: views/js/test/previewer/plugins/navigation/submit/test.js [hunk 6]
```diff
@@ -40,6 +40,36 @@ define([
 ) {
     'use strict';
 
+    const runnerConfig = {
+        serviceCallId : 'foo',
+        providers : {
+            runner: {
+                'id': 'qtiItemPreviewer',
```
Comment (btamas) [inline views/js/test/previewer/plugins/navigation/submit/test.js]: Useless quotes of object key names.

File: views/js/test/previewer/plugins/controls/close/test.js [hunk 7]
```diff
@@ -38,6 +38,36 @@ define([
 ) {
```
Comment (btamas) [inline views/js/test/previewer/plugins/controls/close/test.js]: `core/promise` is not necessary anymore.

File: views/js/test/previewer/plugins/navigation/submit/test.js [hunk 8]
```diff
@@ -40,6 +40,36 @@ define([
 ) {
```
Comment (btamas) [inline views/js/test/previewer/plugins/navigation/submit/test.js]: `core/promise` is not necessary anymore.

File: views/js/test/previewer/runner/test.js [hunk 9]
```diff
@@ -222,7 +221,7 @@ define([
             })
             .on('ready', function(runner) {
                 assert.equal($container.children().length, 1, 'The previewer has been rendered');
-                runner.destroy();
+                setTimeout(() => runner.destroy(), 250);
```
Comment (btamas) [inline views/js/test/previewer/runner/test.js]: Why do you need the timeout?

File: views/js/test/previewer/runner/test.js [hunk 10]
```diff
@@ -22,14 +22,34 @@
...
+        runner : {
+            id: 'qtiItemPreviewer',
+            module: 'taoQtiTestPreviewer/previewer/provider/item/item',
+            bundle: 'taoQtiTestPreviewer/loader/qtiPreviewer.min',
+            category: 'previewer'
+        },
+        proxy: {
+            id: 'qti',
+            module: 'taoQtiTest/runner/proxy/qtiServiceProxy',
+            bundle: 'taoQtiTest/loader/qtiTestRunner.min',
```
Comment (jsconan) [inline views/js/test/previewer/runner/test.js]: This should be instead:
```
id: 'qtiItemPreviewerProxy',
module: 'taoQtiTestPreviewer/previewer/proxy/item',
bundle: 'taoQtiTestPreviewer/loader/qtiPreviewer.min',
```

File: views/js/previewer/proxy/item.js [hunk 11]
```diff
@@ -40,6 +39,8 @@ define([
      */
     return {
 
+        name : 'item-preview',
```
Comment (jsconan) [inline views/js/previewer/proxy/item.js]: Maybe should it be the same name as used elsewhere: `qtiItemPreviewerProxy`

File: views/js/previewer/adapter/item/qtiItem.js [hunk 12]
```diff
@@ -74,22 +64,48 @@ define([
...
+                        "id": "qti",
+                        "module": "taoQtiTest/runner/proxy/qtiServiceProxy",
```
[the remark this answers is not in the document]
Comment (krampstudio) [reply views/js/previewer/adapter/item/qtiItem.js]: my bad, there's another PR on taoQtiTest that adds a name property to the other proxies

File: views/js/test/previewer/runner/test.js (hunk 10 above)
Comment (krampstudio) [reply views/js/test/previewer/runner/test.js]: Does it matter ? I can replace it but I thought that "any" provider can be set without breaking the test

File: scripts/update/Updater.php (hunk 1 above)
Comment (krampstudio) [reply scripts/update/Updater.php]: my bad

File: views/js/test/previewer/runner/test.js (hunk 10 above)
Comment (jsconan) [reply views/js/test/previewer/runner/test.js]: I'm afraid it matters indeed. When I run the tests, it blocks on every test that reference the wrong proxy.

Comment (krampstudio) [reply views/js/test/previewer/runner/test.js]: I know why, I was still loading the correct proxy on https://github.com/oat-sa/extension-tao-testqti-previewer/pull/25/commits/ccde0e2e140ed2e8f9c3e2ab5fb65776654d6790#diff-a974ff9e0fa2cd0d2806828835f6f877L60
Now it's fixed

**Pull request merged**